### PR TITLE
Drop extension_ prefix for extension methods

### PR DESCRIPTION
**Pull request by odersky**
Follow scheme described in https://contributors.scala-lang.org/t/two-proposed-changes-for-extension-methods/4589/25?u=odersky

### CODE REVIEW

**Comment by odersky** [comment]
test performance please

**Comment by odersky** [comment]
I believe performance will improve once we get rid of the double tests that have to account for old as well as new naming schemes for extension methods. We can get rid of that once we are sure that all dependencies are compiled with the new  version.

**Comment by odersky** [comment]
@bishabosha It would be good to get this in quickly. It's been an absolute pain to rebase. Thanks!

File: docs/docs/reference/contextual/extension-methods.md [hunk 1]
```diff
@@ -218,9 +227,9 @@ The precise rules for resolving a selection to an extension method are as follow
 Assume a selection `e.m[Ts]` where `m` is not a member of `e`, where the type arguments `[Ts]` are optional, and where `T` is the expected type.
 The following two rewritings are tried in order:
 
- 1. The selection is rewritten to `extension_m[Ts](e)`.
+ 1. The selection is rewritten to m[Ts](e)`.
```
Comment (bishabosha) [inline docs/docs/reference/contextual/extension-methods.md:230]:
```suggestion
 1. The selection is rewritten to `m[Ts](e)`.
```

File: tests/run/extension-override.scala [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    def len: Int = s.length
+
+object B extends A:
+  extension (s: String)
+    override def len: Int = s.length + 1
+
+@main def Test =
+  import B._
+  assert("abc".len == 4)
+
```
Comment (bishabosha) [inline tests/run/extension-override.scala:12]: I'm not sure how much of an issue this should be, but this import suggestions error masks the fact that the real error is that `B.len` is an illegal override, as we can see I am importing `B.len` in Test
```scala
scala> class A:                                                                                                         
     |   extension (s: String)
     |     def len: Int = s.length
     | 
     | object B extends A:
     |   def len(s: String): Int = s.length + 1
     | 
     | @main def Test =
     |    import B.len
     |    assert("abc".len == 4)
     | 
10 |   assert("abc".len == 4)
   |          ^^^^^^^^^
   |value len is not a member of String, but could be made available as an extension method.
   |
   |The following import might fix the problem:
   |
   |  import B.len
   |
```

**Review by bishabosha — APPROVED** [review]
otherwise LGTM

File: tests/run/extension-override.scala (hunk 2 above)
Comment (odersky) [reply tests/run/extension-override.scala:12]: I think that could be a possible improvement to the import suggestions. I.e. don't make a suggestion if the name is already in scope. Maybe open an issue?

Comment (bishabosha) [reply tests/run/extension-override.scala:12]: opened https://github.com/lampepfl/dotty/issues/10231

**Pull request merged**